### PR TITLE
Move mxnet-dependent code to gluonts.mx

### PR DESCRIPTION
**Pull request by lostella**
*Issue #, if available:* Related to #1181 

*Summary:* This makes sure there is no occurrence of `import mxnet`, `from mxnet`, `from mx`, `import gluonts.mx`, `from gluonts.mx` anywhere in `src`, with the possible exception of modules from `src/gluonts/mx`, `src/gluonts/model`, `src/gluonts/nursery`, and other modules which are used to keep backward compatible (but deprecated) import paths.

*Changes:*

* introduced `skip_encoding` in `gluonts.core.serde`, to allow decoupling `validated` from mxnet dependency, and adapted `validated` accordingly
* simplified `erf`, `erfinv` to work on numpy arrays only (mxnet has it and we were falling back to it anyway)

*Moving code:*

* moving from `gluonts.model` -> `gluonts.mx.model`: `estimator.GluonEstimator`, `forecast.DistributionForecast`
* moving from `gluonts.core.component` -> `gluonts.mx.component`: `equals_parameter_dict`, `equals_representable_block`
* moving from `gluonts.model.common` -> `gluonts.mx`: `Tensor`
* moving `gluonts.support.linalg_util` -> `gluonts.mx.linalg_util`
* moving from `gluonts.support.util` -> `gluonts.mx.util`: `_broadcast_param`, `make_nd_diag`, `cumsum`, `export_repr_block`, `copy_parameters`, `get_hybrid_forward_input_names`, `weighted_average`, `export_symb_block`, `get_hybrid_forward_input_names`, `hybrid_block_to_symbol_block`, `import_repr_block`, `import_symb_block`, `HybridContext`, `assert_shape`

*Other changes:*

* adjust imports to reflect the above changes
* apply isort
* apply black

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jaheba** [comment]
Can you re-order the imports so that the `gluonts.mx`-lines are together?

File: src/gluonts/core/serde/_base.py [hunk 1]
```diff
@@ -81,6 +80,11 @@ class Kind(str, Enum):
     Stateful = "stateful"
 
 
+@singledispatch
+def skip_encoding(v: Any) -> bool:
```
Comment (jaheba) [inline src/gluonts/core/serde/_base.py:84]: Can we add a doc-string?

File: src/gluonts/model/canonical/_network.py [hunk 2]
```diff
@@ -15,11 +15,11 @@
 from mxnet.gluon import HybridBlock
 
 from gluonts.core.component import validated
-from gluonts.model.common import Tensor
 
 # First-party imports
 from gluonts.mx.block.feature import FeatureEmbedder
 from gluonts.mx.block.scaler import MeanScaler
+from gluonts.mx.common import Tensor
```
Comment (jaheba) [inline src/gluonts/model/canonical/_network.py:22]: Not sure I like this `common` thing. (Same with dataset)

Couldn't this be `from gluonts.mx import Tensor`?

**Review by jaheba — DISMISSED** [review]

File: src/gluonts/model/canonical/_network.py (hunk 2 above)
Comment (lostella) [reply src/gluonts/model/canonical/_network.py:22]: sounds good 👍

File: src/gluonts/transform/convert.py [hunk 3]
```diff
@@ -679,7 +678,7 @@ def _fill(target: np.ndarray, expected_length: int) -> np.ndarray:
 
 
 def cdf_to_gaussian_forward_transform(
-    input_batch: DataEntry, outputs: Tensor
+    input_batch: DataEntry, outputs: np.ndarray
```
Comment (lostella) [inline src/gluonts/transform/convert.py:681]: @mbohlkeschneider could you check that these changes to the CDF transform are OK?

Comment (mbohlkeschneider) [reply src/gluonts/transform/convert.py:681]: Should be fine.

**Review by mbohlkeschneider — DISMISSED** [review]
Let's do it.

**Review by jaheba — APPROVED** [review]

**Review by jaheba — APPROVED** [review]

**Pull request merged**